### PR TITLE
refactor:  API response

### PR DESCRIPTION
**Pull request by pranalidhanavade**
## What ##

1) API response refactoring for the following APIs in Users module:
- Fetch login user details.
- Fetch organization invitations.

2) Checked API response, dto validations and promise return type, for all API's in user module.

### CODE REVIEW

File: libs/user-activity/interface/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface user_Activity {
```
Comment (nishad-ayanworks) [inline libs/user-activity/interface/index.ts:1]: Kindly use the best practice to declare a interface name.

**Comment by nishad-ayanworks** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=credebl_platform&pullRequest=344) **Quality Gate passed**
> The SonarCloud Quality Gate passed, but some issues were introduced.
> 
> [1 New issue](https://sonarcloud.io/project/issues?id=credebl_platform&pullRequest=344&resolved=false&inNewCodePeriod=true) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=credebl_platform&pullRequest=344&resolved=false&inNewCodePeriod=true) No data about Coverage [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=credebl_platform&pullRequest=344&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=credebl_platform&pullRequest=344)

@pranalidhanavade Check the sonar cloud issue.

**Comment by nishad-ayanworks** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=credebl_platform&pullRequest=344) **Quality Gate passed**
> The SonarCloud Quality Gate passed, but some issues were introduced.
> 
> [1 New issue](https://sonarcloud.io/project/issues?id=credebl_platform&pullRequest=344&resolved=false&inNewCodePeriod=true) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=credebl_platform&pullRequest=344&resolved=false&inNewCodePeriod=true) No data about Coverage [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=credebl_platform&pullRequest=344&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=credebl_platform&pullRequest=344)

@pranalidhanavade Check the sonar cloud issue

File: libs/user-activity/interface/index.ts (hunk 1 above)
Comment (pranalidhanavade) [reply libs/user-activity/interface/index.ts]: Done

File: apps/user/interfaces/user.interface.ts [hunk 2]
```diff
@@ -1,75 +1,107 @@
...
-    username?: string,
-    email?: string,
-    firstName?: string,
-    lastName?: string,
-    isEmailVerified?: boolean,
-    clientId?: string,
-    clientSecret?: string,
-    supabaseUserId?: string,
-    userOrgRoles?: object
+export  interface UInvitation {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:1]: Please use meaningful names for interface.

File: apps/user/interfaces/user.interface.ts [hunk 3]
```diff
@@ -1,75 +1,107 @@
...
+  orgRoles?: OrgRole[];
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:2]: Every parameter in an interface can not be optional. Please have mandatory parameters by removing "?"

File: apps/user/interfaces/user.interface.ts [hunk 4]
```diff
@@ -1,75 +1,107 @@
...
+  name: string;
+  description: string;
 }
 
-export interface UserWhereInput{
-    email?: string
-}
-export interface UpdateUserProfile {
-    id: string,
+export interface UserI {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:20]: Please use meaningful names for interface.

File: apps/user/interfaces/user.interface.ts [hunk 5]
```diff
@@ -1,75 +1,107 @@
...
+    id?: string;
+    username?: string;
+    email?: string;
+    firstName?: string;
+    lastName?: string;
+    supabaseUserId?: string;
+    userOrgRoles?: object;
+  }
+  
+  export interface InvitationsI {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:30]: Please use meaningful names for interface.

File: apps/user/interfaces/user.interface.ts [hunk 6]
```diff
@@ -1,75 +1,107 @@
...
+    firstName: string;
+    lastName: string;
+    isPasskey: boolean;
+  }
+  
+  export interface AddPasskeyDetails {
+    password: string;
+  }
+  
+  export interface UserWhereUniqueInput {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:58]: Please use meaningful names for interface.

File: apps/user/interfaces/user.interface.ts [hunk 7]
```diff
@@ -1,75 +1,107 @@
...
+    id?: string;
+  }
+  
+  export interface UserWhereInput {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:62]: Please use meaningful names for interface.

File: apps/user/interfaces/user.interface.ts [hunk 8]
```diff
@@ -1,75 +1,107 @@
...
-    isPublic: boolean,
-}
-export interface PlatformSettingsI {
-    externalIp: string,
-    lastInternalId: string,
+    firstName: string;
+    lastName: string;
+    isPublic: boolean;
+  }
+  export interface PlatformSettingsI {
```
Comment (vivekayanworks) [inline apps/user/interfaces/user.interface.ts:72]: Let's follow consistency in Interface naming convention. Please remove "I" at the end of interface.

File: libs/user-activity/interface/index.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+export interface User_Activity {
```
Comment (vivekayanworks) [inline libs/user-activity/interface/index.ts:1]: Let's follow consistency in Interface naming convention.

**Comment by ajile-in** [comment]
@pranalidhanavade , @nishad-ayanworks  Let's stick to standard naming conventions.

File: apps/user/interfaces/user.interface.ts (hunk 2 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 4 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 5 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 6 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 7 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 8 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

File: libs/user-activity/interface/index.ts (hunk 9 above)
Comment (pranalidhanavade) [reply libs/user-activity/interface/index.ts]: Resolved.

File: apps/user/interfaces/user.interface.ts (hunk 3 above)
Comment (pranalidhanavade) [reply apps/user/interfaces/user.interface.ts]: Resolved.

**Review by vivekayanworks — APPROVED** [review]

**Pull request merged**